tools/Customization/DevHome.Customization/ViewModels/VirtualizationFeatureManagementViewModel.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.WinUI;
+using CommunityToolkit.WinUI.Behaviors;
+using DevHome.Common.Extensions;
+using DevHome.Common.Helpers;
+using DevHome.Common.Models;
+using DevHome.Common.Scripts;
+using DevHome.Common.Services;
+using DevHome.Customization.Views;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Serilog;
+
+namespace DevHome.Customization.ViewModels;
+
+public partial class VirtualizationFeatureManagementViewModel : ObservableObject
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(VirtualizationFeatureManagementViewModel));
+
+    private readonly StringResource _stringResource;
+
+    private readonly bool _isUserAdministrator = WindowsIdentityHelper.IsUserAdministrator();
+
+    private readonly Dictionary<string, bool> _initialFeatureEnabledStates = new();
+
+    private readonly Window _window;
+
+    private readonly ModifyFeaturesDialog _modifyFeaturesDialog;
+
+    private StackedNotificationsBehavior? _notificationQueue;
+
+    public IAsyncRelayCommand LoadFeaturesCommand { get; }
+
+    public bool FeaturesLoaded => !_isFirstLoad || !LoadFeaturesCommand.IsRunning;
+
+    public IAsyncRelayCommand ApplyChangesCommand { get; }
+
+    public bool ChangesCanBeApplied => HasFeatureChanges && !ApplyChangesCommand.IsRunning;
+
+    public ObservableCollection<Breadcrumb> Breadcrumbs { get; }
+
+    public ObservableCollection<WindowsOptionalFeatureState> Features { get; } = new();
+
+    public bool HasFeatureChanges => _isUserAdministrator && FeaturesLoaded && Features.Any(f => f.HasChanged);
+
+    public bool CanDismissNotifications => _isUserAdministrator;
+
+    private bool _restartNeeded;
+
+    private bool _isFirstLoad;
+
+    public VirtualizationFeatureManagementViewModel(Window window)
+    {
+        _stringResource = new StringResource("DevHome.Customization.pri", "DevHome.Customization/Resources");
+        _window = window;
+        _isFirstLoad = true;
+
+        Breadcrumbs =
+        [
+            new(_stringResource.GetLocalized("MainPage_Header"), typeof(MainPageViewModel).FullName!),
+            new(_stringResource.GetLocalized("VirtualizationFeatureManagement_Header"), typeof(VirtualizationFeatureManagementViewModel).FullName!)
+        ];
+
+        LoadFeaturesCommand = new AsyncRelayCommand(LoadFeaturesAsync);
+        LoadFeaturesCommand.PropertyChanged += async (s, e) =>
+        {
+            if (e.PropertyName == nameof(LoadFeaturesCommand.IsRunning))
+            {
+                await OnFeaturesChanged();
+            }
+        };

```

Comment:
is this how it's supposed to be done :P ?

Reply:
The problem with IsRunning, is that I need the inverse here and building a converter to do teh negation wasn't working out very well. Seems a pattern a lot of products use rather than negation converters was explicit properties FeaturesLoaded for this particular case. It makes it clear in the XAML what connection is and makes it explicit in the code.